tests/integration/targets/dns_auth_nsg/tasks/main.yml

```diff
@@ -0,0 +1,215 @@
+#        state: absent
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg is changed
+#          - auth_nsg_info is not failed
+#          - auth_nsg_info.objects | length == 0
+#
+#    - name: Delete the Auth NSG (idempotent)
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        state: absent
+#      register: auth_nsg
+#    - assert:
+#        that:
+#          - auth_nsg is not changed
+#          - auth_nsg is not failed
+#
+#    - name: Create an Auth NSG with a comment
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        comment: "Test Comment"
+#        state: present
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg_info is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].id == auth_nsg.id
+#          - auth_nsg_info.objects[0].comment == "Test Comment"
+#
+#    - name: Create Auth NSG with external primary
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        external_primaries:
+#          - address: "1.1.1.1"
+#            fqdn: "a.com."
+#            type: "primary"
+#        state: present
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].external_primaries | length == 1
+#          - auth_nsg_info.objects[0].external_primaries[0].address == "1.1.1.1"
+#          - auth_nsg_info.objects[0].external_primaries[0].fqdn == "a.com."
+#          - auth_nsg_info.objects[0].external_primaries[0].type == "primary"
+#
+#    - name: Create an Auth NSG with tags
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        comment: "Test Comment"
+#        tags:
+#          location: "site-1"
+#        state: present
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg_info is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].id == auth_nsg.id
+#          - auth_nsg_info.objects[0].tags.location == "site-1"
+
+    - name: Create the Primary Auth NSG
```

Comment:
We can use Primary NSG to be the one that is created using the task with Tags above , no need to create an additional NSG !